lib/compress/zstd_ldm.c

```diff
@@ -154,6 +154,24 @@ static size_t ZSTD_ldm_countBackwardsMatch(
     return matchLength;
 }
 
+/** ZSTD_ldm_countBackwardsMatch_2segments() :
+ *  Returns the number of bytes that match backwards from pMatch,
+ *  even with the backwards match spanning 2 different segments.
+ *
+ *  On reaching `pMatchBase`, start counting from mEnd */
+static size_t ZSTD_ldm_countBackwardsMatch_2segments(
+            const BYTE* pIn, const BYTE* pAnchor,
+            const BYTE* pMatch, const BYTE* pMatchBase, const BYTE* dictStart, const BYTE* mEnd)
```

Comment:
nit: `mEnd` isn't the right name for this variable. `dictEnd` maybe?

Reply:
I agree, the variable names are very confusing. I'm gonna go through and change some stuff for clarity